client/src/Component/SMSModal.js

```diff
@@ -1,23 +1,54 @@
-/* eslint-disable jsx-a11y/label-has-for */
+
 import { useState } from "react";
 import Modal from "react-bootstrap/Modal";
 import Button from "react-bootstrap/Button";
 
+
+
+
 function SMSModal(props) {
     const [show, setShow] = useState(false);
     const hideModal = () => setShow(false);
     const showModal = () => setShow(true);
 
+    let localMessage = localStorage.getItem("message");
+    if (localMessage === null) {
+      localMessage = "Hi, this is Bikes For Refugees. Bikes are now available for you to pick up. Please click LINK to choose a pick up date. For any questions, please contact 07900000000.";
+    }
+    const [message, setMessage] = useState(localMessage);
+
+    const handleChange = (e) => {
+      localStorage.setItem("message", e.target.value);
+      setMessage(e.target.value);
+    };
+
 
     function handleSubmit () {
-      console.log(props.checkedCheckboxes);
-      hideModal();
-      }
+
+      const newMessageRequest = {
+          ids : props.selectedClients,
+          message: message,
+        };
+
+      fetch ("/api/send-messages", {
+          method: "POST",
+          headers: {
+            "Content-Type": "application/json",
+          },
+          body: JSON.stringify(newMessageRequest),
+        })
+        .then((response) => {
+          hideModal();
```

Comment:
Let's update the code to only do this `then` on success, not on error too

Reply:
This is done, however I'm not sure if I've maybe added more code than I need to so look forward to seeing what you think!